src/main/java/uk/gov/companieshouse/officer/delta/processor/apiclient/InternalApiClientFactory.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.companieshouse.officer.delta.processor.apiclient;
+
+import java.util.function.Supplier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.InternalApiClient;
+import uk.gov.companieshouse.api.http.ApiKeyHttpClient;
+import uk.gov.companieshouse.officer.delta.processor.logging.DataMapHolder;
+
+@Component
+public class InternalApiClientFactory implements Supplier<InternalApiClient> {
+
+    private final String apiKey;
+    private final String apiUrl;
+    private final String paymentsApiUrl;
+    private final String internalApiUrl;
+
+    public InternalApiClientFactory(@Value("${chs.internal.api.key}") String apiKey,
+            @Value("${api.url}") String apiUrl,
+            @Value("${payments.api.url}") String paymentsApiUrl,
```

Comment:
I think the payments URL and the internal URL can both be removed from here and app properties/config files. We only ever call company appointments API using the base API URL.